btree/btree.c

```diff
@@ -2681,6 +2685,23 @@ static void m0_btree_put_split_and_find(struct nd *l_alloc , struct nd *l_node,
 	diff = m0_bufvec_cursor_cmp(&cur_1, &cur_2);
 	tgt->s_node = diff < 0 ? l_slot.s_node : r_slot.s_node;
 
+	/**
+	 * Corner case: If given record needs to be inseted at internal left
+	 * node and if the key of given record is greater than key at last index
+	 * of left record, initialised tgt->s_idx explicitly, as node_find will
+	 * not compare key with last indexed key.
+	 */
+	if (node_level(tgt->s_node) > 0 && tgt->s_node == l_slot.s_node) {
+		l_slot.s_idx = node_count(l_slot.s_node);
+		l_slot.s_rec = temp_rec;
+		node_key(&l_slot);
+		m0_bufvec_cursor_init(&cur_2, &r_slot.s_rec.r_key.k_data);
```

Comment:
The second parameter should be &l_slot.s_rec.r_key.k_data and not &r_slot.s_rec.r_key.k_data.

Reply:
yes right